day4/main.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func main() {
+
+	if len(os.Args) != 2 {
+		fmt.Println("input file path not provided")
+		return
+	}
+	input := readFileLineByLine(os.Args[1])
+
+	var sum int
+	for _, line := range input {
+		sum += findNumbers(line)
+	}
+	fmt.Println("answer for part 1: ", sum)
+
+	fmt.Println("answer for part 2: ", findNumberOfCards(input))
+}
+
+func readFileLineByLine(path string) []string {
+	file, err := os.Open(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	var output []string
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		output = append(output, scanner.Text())
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+
+	return output
+}
+
+func findNumbers(input string) int {
```

Comment:
you could return here the result and the matches to use this in findNumberOfCards
`func findNumbers(input string) (int, int)`